### PR TITLE
MIR-924 abstract html handling is wrong

### PR DESCRIPTION
**Pull request by sebhofmann**
[Link to jira](https://mycore.atlassian.net/browse/MIR-924).

### CODE REVIEW

**Comment by fluetze** [comment]
Is "allowHTML" a good name for the property? HTML was also allowed before, there was no need to set the property, and if the property existed in the past, we'd had to set it to "false" to "enable" html. Somewhat ugly. What exactly does the property do?

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with 71a73a61b4f3292e6cf91c1349cb2c405f52cec3! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/2808-pull_request-502

File: mir-module/src/main/java/org/mycore/mir/editor/MIREditorTools.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package org.mycore.mir.editor;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Entities;
+import org.w3c.dom.Node;
```
Comment (yagee-de) [inline mir-module/src/main/java/org/mycore/mir/editor/MIREditorTools.java:6]: Codacy found an issue: [Avoid unused imports such as 'org.w3c.dom.Node'](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- mir-module/src/main/java/org/mycore/mir/editor/MIRPostProcessor.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIREditorTools.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRUnescapeResolver.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770&bid=19106470)

File: mir-module/src/main/java/org/mycore/mir/editor/MIREditorTools.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+public class MIREditorTools {
```
Comment (yagee-de) [inline mir-module/src/main/java/org/mycore/mir/editor/MIREditorTools.java:8]: Codacy found an issue: [The utility class name 'MIREditorTools' doesn't match '[A-Z][a-zA-Z0-9]+(Utils?|Helper|Constants)'](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

File: mir-module/src/main/java/org/mycore/mir/editor/MIREditorTools.java (hunk 1 above)
Comment (yagee-de) [inline mir-module/src/main/java/org/mycore/mir/editor/MIREditorTools.java:6]: Codacy found an issue: [Unused import - org.w3c.dom.Node.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with 74f7b0145f46d8da56cf8ad4b098deb50cf7a923! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/2810-pull_request-502

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mir-module/src/main/java/org/mycore/mir/editor/MIREditorUtils.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRPostProcessor.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRUnescapeResolver.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770&bid=19106470)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with b890a361197bf51f68ae46d0aa17c81b3c4ac57b! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/2812-pull_request-502

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mir-module/src/main/java/org/mycore/mir/editor/MIRUnescapeResolver.java  5
- mir-module/src/main/java/org/mycore/mir/editor/MIRPostProcessor.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIREditorUtils.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770&bid=19106470)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with 4c0ad238db22fd014b6265e46116ae620030ba70! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/2814-pull_request-502

File: mir-module/GruntFile.js [hunk 3]
```diff
@@ -71,15 +72,25 @@ module.exports = function(grunt) {
...
+          '<%= globalConfig.moduleDirectory %>/target/classes/META-INF/assets/ckeditor4/config.min.js': '<%= globalConfig.moduleDirectory %>/src/main/resources/META-INF/resources/js/ckeditor-config.js'
         }
       }
     },
   });
+  grunt.registerTask('copy', function() {
+    const source = getAbsoluteDir(grunt.option('moduleDirectory'))+'/src/main/resources/META-INF/resources/js/ckeditor-config.js';
+    const target = getAbsoluteDir(grunt.option('assetsDirectory')) + '/ckeditor4/config.js';
+    fs.copyFile(source, target, (err) => {
+      if (err) throw err;
```
Comment (yagee-de) [inline mir-module/GruntFile.js:85]: Codacy found an issue: [Expected { after 'if' condition.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

Comment (yagee-de) [inline mir-module/GruntFile.js:85]: Codacy found an issue: [Avoid using if statements without curly braces](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

File: mir-module/GruntFile.js [hunk 4]
```diff
@@ -71,15 +72,25 @@ module.exports = function(grunt) {
...
+      grunt.log('ckeditor config copied!');
+    });
+  })
   grunt.registerTask('none', function() {
   });
   grunt.registerTask('default', 'build assets directory', function() {
     grunt.task.run('npmcopy');
+    grunt.task.run('copy')
```
Comment (yagee-de) [inline mir-module/GruntFile.js:93]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- mir-module/src/main/java/org/mycore/mir/editor/MIRUnescapeResolver.java  5
- mir-module/src/main/java/org/mycore/mir/editor/MIREditorUtils.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRPostProcessor.java  1
- mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770&bid=19106470)

File: mir-module/GruntFile.js [hunk 5]
```diff
@@ -71,15 +72,25 @@ module.exports = function(grunt) {
...
     },
   });
+  grunt.registerTask('copy', function() {
+    const source = getAbsoluteDir(grunt.option('moduleDirectory'))+'/src/main/resources/META-INF/resources/js/ckeditor-config.js';
+    const target = getAbsoluteDir(grunt.option('assetsDirectory')) + '/ckeditor4/config.js';
+    fs.copyFile(source, target, (err) => {
+      if (err) throw err;
+      grunt.log('ckeditor config copied!');
+    });
+  })
```
Comment (yagee-de) [inline mir-module/GruntFile.js:88]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

File: mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+CKEDITOR.editorConfig = function( config ) {
```
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js:1]: Codacy found an issue: ['CKEDITOR' is not defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with 53c8878772490b77bc069f28bde906502ac12059! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/2816-pull_request-502

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- mir-module/src/main/java/org/mycore/mir/editor/MIRPostProcessor.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIREditorUtils.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRUnescapeResolver.java  5
- mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770&bid=19106470)

File: mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js (hunk 6 above)
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js:1]: Codacy found an issue: ['CKEDITOR' is not defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with f3aec775ed3d2794b260af8c79f6c773e54d3d99! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/2818-pull_request-502

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRUnescapeResolver.java  5
- mir-module/src/main/java/org/mycore/mir/editor/MIREditorUtils.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRPostProcessor.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770&bid=19106470)

File: mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js (hunk 6 above)
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js:1]: Codacy found an issue: ['CKEDITOR' is not defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with 48dd8eb3ce6876fca299d89e46e2c18c4be539e1! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/2820-pull_request-502

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- mir-module/src/main/java/org/mycore/mir/editor/MIREditorUtils.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRUnescapeResolver.java  5
- mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRPostProcessor.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770&bid=19106470)

File: mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js (hunk 6 above)
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js:1]: Codacy found an issue: ['CKEDITOR' is not defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

File: mir-it/src/test/java/org/mycore/mir/it/controller/MIREditorController.java [hunk 7]
```diff
@@ -30,6 +31,13 @@ protected void setTextAreaText(String childElementName, String text) {
         driver.waitAndFindElement(By.xpath(".//textarea[contains(@name,'" + childElementName + "')]")).sendKeys(text);
     }
 
+    protected void setHTMLAreaText(String childElementName, String text) {
+        final WebElement iframe = driver.waitAndFindElement(By.xpath(".//div[contains(@id,'" + childElementName + "') and contains(@class, 'cke')]//iframe"));
```
Comment (yagee-de) [inline mir-it/src/test/java/org/mycore/mir/it/controller/MIREditorController.java:35]: Codacy found an issue: [Line is longer than 120 characters (found 158).](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with dba5d09d472a380a4c7528193f8376e7a7aff738! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/2822-pull_request-502

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- mir-module/src/main/java/org/mycore/mir/editor/MIREditorUtils.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRUnescapeResolver.java  5
- mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRPostProcessor.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770&bid=19106470)

File: mir-it/src/test/java/org/mycore/mir/it/controller/MIREditorController.java (hunk 7 above)
Comment (yagee-de) [inline mir-it/src/test/java/org/mycore/mir/it/controller/MIREditorController.java:35]: Codacy found an issue: [Line is longer than 120 characters (found 158).](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

File: mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js (hunk 6 above)
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js:1]: Codacy found an issue: ['CKEDITOR' is not defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with 2ee11be4d5896d53e77cb75726e9d63808ff1016! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/2823-pull_request-502

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js  1
- mir-module/src/main/java/org/mycore/mir/editor/MIREditorUtils.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRUnescapeResolver.java  5
- mir-module/src/main/java/org/mycore/mir/editor/MIRPostProcessor.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770&bid=19106470)

File: mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js (hunk 6 above)
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js:1]: Codacy found an issue: ['CKEDITOR' is not defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

File: mir-it/src/test/java/org/mycore/mir/it/controller/MIREditorController.java [hunk 8]
```diff
@@ -30,6 +31,14 @@ protected void setTextAreaText(String childElementName, String text) {
...
```
Comment (yagee-de) [inline mir-it/src/test/java/org/mycore/mir/it/controller/MIREditorController.java:35]: Codacy found an issue: [Line is longer than 120 characters (found 158).](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- mir-module/src/main/java/org/mycore/mir/editor/MIREditorUtils.java  1
- mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRUnescapeResolver.java  5
- mir-module/src/main/java/org/mycore/mir/editor/MIRPostProcessor.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770&bid=19106470)

File: mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js (hunk 6 above)
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js:1]: Codacy found an issue: ['CKEDITOR' is not defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- mir-module/src/main/java/org/mycore/mir/editor/MIRPostProcessor.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIREditorUtils.java  1
- mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRUnescapeResolver.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770&bid=19106470)

File: mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
```
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js:1]: Codacy found an issue: ['CKEDITOR' is not defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRUnescapeResolver.java  5
- mir-module/src/main/java/org/mycore/mir/editor/MIREditorUtils.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRPostProcessor.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770&bid=19106470)

File: mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js (hunk 9 above)
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js:1]: Codacy found an issue: ['CKEDITOR' is not defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- mir-module/src/main/java/org/mycore/mir/editor/MIRPostProcessor.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRUnescapeResolver.java  5
- mir-module/src/main/java/org/mycore/mir/editor/MIREditorUtils.java  1
- mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770&bid=19106470)

File: mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js (hunk 9 above)
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js:1]: Codacy found an issue: ['CKEDITOR' is not defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRUnescapeResolver.java  5
- mir-module/src/main/java/org/mycore/mir/editor/MIREditorUtils.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRPostProcessor.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770&bid=19106470)

File: mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js (hunk 9 above)
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js:1]: Codacy found an issue: ['CKEDITOR' is not defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- mir-module/src/main/java/org/mycore/mir/editor/MIRPostProcessor.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIREditorUtils.java  1
- mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRUnescapeResolver.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770&bid=19106470)

File: mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js (hunk 9 above)
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js:1]: Codacy found an issue: ['CKEDITOR' is not defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRUnescapeResolver.java  5
- mir-module/src/main/java/org/mycore/mir/editor/MIREditorUtils.java  1
- mir-module/src/main/java/org/mycore/mir/editor/MIRPostProcessor.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770&bid=19106470)

File: mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
```
Comment (yagee-de) [inline mir-module/src/main/resources/META-INF/resources/js/ckeditor-config.js:1]: Codacy found an issue: ['CKEDITOR' is not defined.](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=5834770)

**Pull request merged**